src/utils/video.ts

```diff
@@ -10,3 +10,39 @@ export const formatVideoViewsAndDate = (
   const formattedViews = views !== null && (fullViews ? formatNumber(views) : formatNumberShort(views))
   return formattedViews ? `${formattedViews} views • ${formattedDate}` : formattedDate
 }
+
+export type VideoMetadata = {
+  mimeType: string
+  sizeInBytes: number
+  height: number
+  width: number
+  duration: number
+}
+export const getVideoMetadata = async (file?: File): Promise<{ metadata: VideoMetadata }> => {
```

Comment:
I don't think there's a point in wrapping the data in additional object, let's just return `Promise<VideoMetadata>